chatty-paper/src/main/kotlin/com/mineinabyss/chatty/listeners/ChattyProxyListener.kt

```diff
@@ -50,46 +50,67 @@ class ChattyProxyListener : PluginMessageListener {
                 ChannelType.GLOBAL -> onlinePlayers
                 ChannelType.RADIUS -> canSpy
                 ChannelType.PERMISSION -> onlinePlayers.filter { it.hasPermission(channel.permission) || it in canSpy }
-                ChannelType.PRIVATE -> onlinePlayers.filter { it.getChannelFromPlayer() == channel || it in canSpy }
+                ChannelType.PRIVATE -> onlinePlayers.filter {
+                    it.toGeary().get<ChannelData>()?.channel == channel || it in canSpy
+                }
             }.forEach { it.sendMessage(proxyMessage.miniMsg()) }
         }
 
 
-        if (!chatty.config.proxy.sendProxyMessagesToDiscord ||
-            channel?.discordsrv != true || !chatty.isDiscordSRVLoaded
-        ) return
+        if (chatty.config.proxy.sendProxyMessagesToDiscord
+            && channel?.discordsrv == true
+            && chatty.isDiscordSRVLoaded
+        ) {
+            sendToDiscord(proxyMessage, senderName, channel, channelFormat)
+        }
+
+    }
 
+    fun sendToDiscord(proxyMessage: String, senderName: String, channel: ChattyChannel, channelFormat: String) {
         val reserializer = DiscordSRV.config().getBoolean("Experiment_MCDiscordReserializer_ToDiscord")
-        val discordChannel = DiscordSRV.getPlugin().getDestinationTextChannelForGameChannelName(chatty.config.proxy.discordSrvChannelID)
+        val discordChannel = DiscordSRV.getPlugin()
+            .getDestinationTextChannelForGameChannelName(chatty.config.proxy.discordSrvChannelID)
 
         when {
             discordChannel == null -> {
-                DiscordSRV.debug(Debug.MINECRAFT_TO_DISCORD,
-                    "Failed to find Discord channel to forward message from game channel $channel")
+                DiscordSRV.debug(
+                    Debug.MINECRAFT_TO_DISCORD,
+                    "Failed to find Discord channel to forward message from game channel $channel"
+                )
             }
+
             !DiscordUtil.checkPermission(discordChannel.guild, Permission.MANAGE_WEBHOOKS) ->
                 DiscordSRV.error("Couldn't deliver chat message as webhook because the bot lacks the \"Manage Webhooks\" permission.")
+
             else -> {
                 val discordMessage = proxyMessage.replaceFirst(channelFormat, "")
                     .apply { PlaceholderUtil.replacePlaceholdersToDiscord(this) }
                     .apply { if (!reserializer) MessageUtil.strip(this) }
-                    .apply { if (translateMentions) DiscordUtil.convertMentionsFromNames(this, DiscordSRV.getPlugin().mainGuild) }
+                    .apply {
+                        if (translateMentions) DiscordUtil.convertMentionsFromNames(
+                            this,
+                            DiscordSRV.getPlugin().mainGuild
+                        )
+                    }
```

Comment:
apply wouldnt return the changes so is this even really achieving anything?
think i forgot to tweak this, either way DiscordSRV alternative is still something im looking for so for now this is fine